### DISCUSSION TITLE
Polling stopping during scan process or not completing

### DISCUSSION THREAD

**Discussion by BUGNOTTO** [terjeio/ioSender#326] [General]
Good morning everyone, first forgive me if I can't be clear, my English is weak so I'm using the google translator, I'm from Brazil.

I have the following problem. During the probing process, the IO Sender is crashing, it simply stops or even goes to the end but does not show the vector drawing with the scanned points. I'll put some images on the IO sender. Do you have any logs to review? I'm using the latest version 2.0.43.
![sondagem 01](https://github.com/terjeio/ioSender/assets/99672754/ac023166-61eb-41a4-be47-8b2200f90ba4)
![sondagem 02](https://github.com/terjeio/ioSender/assets/99672754/6404d82a-64c1-4bab-a8f8-49cc67b1b114)

NOTE: My probe (milling cutter) is not touching anything and is inside the machining area (Cartesian plane) as per the print image.

![WhatsApp Image 2023-09-04 at 11 43 33 AM](https://github.com/terjeio/ioSender/assets/99672754/5b562c80-687f-4986-95c9-e7211592420f)
![WhatsApp Image 2023-09-04 at 11 43 06 AM (1)](https://github.com/terjeio/ioSender/assets/99672754/2e5c646c-dc0e-4232-b869-32641254ed2b)
![WhatsApp Image 2023-09-04 at 11 43 06 AM](https://github.com/terjeio/ioSender/assets/99672754/f9bd186a-357c-4929-9141-0bc6e9ef0226)

**Comment by BUGNOTTO** [comment]
now until the entire scanning process is finished. however I think that the Io sender was not able to read or store all the points.
![sondagem 03](https://github.com/terjeio/ioSender/assets/99672754/f2fa601f-7c35-47b4-8c0e-c13022f2b823)

look how it looked in the 3D simulation after applying the sounding.

![sondagem 04](https://github.com/terjeio/ioSender/assets/99672754/4d9d855d-893d-4211-a3b3-a5c16865a3eb)

**Comment by BUGNOTTO** [comment]
now on the third try it worked
![sondagem 05](https://github.com/terjeio/ioSender/assets/99672754/421a2c60-abb7-44e1-b15e-cdaf5b632237)
![sondagem 06](https://github.com/terjeio/ioSender/assets/99672754/500489e2-f8d4-4c39-80f7-c3fd70ac49d6)

**Reply by terjeio** [reply]
> however I think that the Io sender was not able to read or store all the points.

You will not be able to apply the transform unless all points were sucessfully aquired.

I am curious about the crash, did you get any messages when it happens?

> Do you have any logs to review?

No, currently only debufg versions writes to a log file. Some logging related to probing can be accessed if you check _Verbose_ mode in the console. FYI a floating console window can be opened via the _File_ menu if you want to see the messages appear while probing.

**Reply by BUGNOTTO** [reply]
worse than that I didn't receive any error message, the cnc simply stops during the probing process, even with the Verbose option checked : /
I checked all the options in this floating window and it still doesn't show an error message.

**Comment by BUGNOTTO** [comment]
![WhatsApp Image 2023-09-05 at 17 53 04](https://github.com/terjeio/ioSender/assets/99672754/4830e4db-8d9c-4fab-aaf9-f6b2ad54fd67)

look what happened to the feed rate :/

**Comment by phil-barrett** [comment]
Wow, that's quite a machine, 257 KPH!

**Reply by BUGNOTTO** [reply]
excuse my ignorance : ) but where is this information ? or what would be KPH?

**Reply by phil-barrett** [reply]
4294696 mm/min is 4.295 KM/min which is 257 KM/Hr (KPH).  Sorry, just making a joke.

**Reply by BUGNOTTO** [reply]
hehehehe I imagined KM/H .... it would be a formula 1 car kkkk .... The funny thing is that when I start the cycle it goes very slowly I think around 1 to 3 mm/min ... very slowly and even I change it the feed rate percentage does not change speed the z axis goes down very slowly

**Comment by BUGNOTTO** [comment]
I'm thinking it's some error or bug in the program. how would i go about getting some error log. Would ?

**Reply by terjeio** [reply]
Check _Verbose_ and _Show all realtime responses_ in the console. If nothing is output then the controller has died?
Does pressing the red "Reset" (Reiniciar) button help?

Paste your controller settings into a comment, from _Help > About > To Clipboard_, it could be that some are way off? FYI I have had a few occasions during development where the controller acted strangely and a settings reset was the only way to get it to behave again. A reset can be done by sending `$RST=*`

**Reply by BUGNOTTO** [reply]
Hello budy, how are you? I didn't understand this second part "Paste your controller settings into a comment, from Help > About > To Clipboard, it could be that some are way off? FYI I have had a few occasions during development where the controller acted strangely and a settings reset was the only way to get it to behave again. A reset can be done by sending $RST=*" 
![erro](https://github.com/terjeio/ioSender/assets/99672754/24e55a98-5c8b-4b92-8f65-b9d3a867c8c8)

Would it be on this screen to paste my controller settings? Where do I find my controller settings? How to use $RST=* ? I didn't understand :/

**Reply by terjeio** [reply]
> Would it be on this screen to paste my controller settings?

They will be copied to the clipboard when you press the _To clipboard_ button. Then give focus to a comment frame and press CTRL+V (unless this has changed in Brasilian Windows) or right click the mouse in the frame and select Paste (or whatever it is called in Brasilian). All the settings will then appear in the frame.

>  How to use $RST=* ? I didn't understand :/

You send commands to the controller from the MDI, just type the command and press enter (or the _Enviar_ button).

![image](https://github.com/terjeio/ioSender/assets/20260062/a455ec5e-4aa4-4460-87df-c0697dcf425e)

Make a backup of your settings first.

**Comment by BUGNOTTO** [comment]
[log.txt](https://github.com/terjeio/ioSender/files/12542756/log.txt)
I generated the log file, what caught my attention is FS:4294967295.0 from the beginning,

**Comment by BUGNOTTO** [comment]
<Run|MPos:0.000,0.000,0.520|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,0.523|Bf:34,1023|FS:4294967295,0|Ov:160,25,100>
<Run|MPos:0.000,0.000,0.526|Bf:34,1023|FS:4294967295,0|Ov:150,25,100>
<Run|MPos:0.000,0.000,0.530|Bf:34,1023|FS:4294967295,0|Ov:130,25,100>
<Run|MPos:0.000,0.000,0.533|Bf:34,1023|FS:4294967295,0|Ov:120,25,100>
<Run|MPos:0.000,0.000,0.536|Bf:34,1023|FS:4294967295,0|Ov:110,25,100>
<Run|MPos:0.000,0.000,0.539|Bf:34,1023|FS:4294967295,0|Ov:90,25,100>
<Run|MPos:0.000,0.000,0.542|Bf:34,1023|FS:4294967295,0|Ov:80,25,100>
<Run|MPos:0.000,0.000,0.545|Bf:34,1023|FS:4294967295,0|Ov:70,25,100>
<Run|MPos:0.000,0.000,0.548|Bf:34,1023|FS:4294967295,0|Ov:50,25,100>
<Run|MPos:0.000,0.000,0.550|Bf:34,1023|FS:4294967295,0|Ov:40,25,100>
<Run|MPos:0.000,0.000,0.553|Bf:34,1023|FS:4294967295,0|Ov:30,25,100>
<Run|MPos:0.000,0.000,0.556|Bf:34,1023|FS:4294967295,0|Ov:10,25,100>
<Run|MPos:0.000,0.000,0.559|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,0.562|Bf:34,1023|FS:4294967295,0|WCO:0.000,0.000,-0.928>
<Run|MPos:0.000,0.000,0.565|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,0.569|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,0.572|Bf:34,1023|FS:4294967295,0>

**Comment by BUGNOTTO** [comment]
next try :/

[G58:0.000,0.000,0.000]
[G59:0.000,0.000,0.000]
[G59.1:0.000,0.000,0.000]
[G59.2:0.000,0.000,0.000]
[G59.3:0.000,0.000,0.000]
[G28:0.000,0.000,0.000]
[G30:0.000,0.000,0.000]
[G92:0.000,0.000,-33.238]
[HOME:0.000,0.000,0.000:0]
[TLO:0.000,0.000,0.000]
[PRB:0.000,0.000,-1.384:1]
[TLR:-1.075]
ok
[GC:G0 G54 G92 G17 G21 G90 G94 G49 G98 G50 M5 M9 T0 F25 S0.]
ok
[G54:0.000,0.000,31.854]
[G55:0.000,0.000,0.000]
[G56:0.000,0.000,0.000]
[G57:0.000,0.000,0.000]
[G58:0.000,0.000,0.000]
[G59:0.000,0.000,0.000]
[G59.1:0.000,0.000,0.000]
[G59.2:0.000,0.000,0.000]
[G59.3:0.000,0.000,0.000]
[G28:0.000,0.000,0.000]
[G30:0.000,0.000,0.000]
[G92:0.000,0.000,-33.238]
[HOME:0.000,0.000,0.000:0]
[TLO:0.000,0.000,0.000]
[PRB:0.000,0.000,-1.384:1]
[TLR:-1.075]
ok
<Idle|MPos:0.000,0.000,0.000|Bf:35,1023|FS:0,0|WCO:0.000,0.000,-1.384|WCS:G54|A:|Sc:|H:0|T:0|TLR:1|FW:grblHAL>
<Idle|MPos:0.000,0.000,0.000|Bf:35,1023|FS:0,0|WCO:0.000,0.000,-1.384>
G90
G90
ok
<Idle|MPos:0.000,0.000,0.000|Bf:35,1023|FS:0,0|WCO:0.000,0.000,-1.384|WCS:G54|A:|Sc:|H:0|T:0|TLR:1|FW:grblHAL>
<Idle|MPos:0.000,0.000,0.000|Bf:35,1023|FS:0,0|WCO:0.000,0.000,-1.384>
<Idle|MPos:0.000,0.000,0.000|Bf:35,1023|FS:0,0|Ov:100,50,100>
<Idle|MPos:0.000,0.000,0.000|Bf:35,1023|FS:0,0|Ov:90,50,100>
<Idle|MPos:0.000,0.000,0.000|Bf:35,1023|FS:0,0|Ov:80,50,100>
<Idle|MPos:0.000,0.000,0.000|Bf:35,1023|FS:0,0|Ov:70,50,100>
<Idle|MPos:0.000,0.000,0.000|Bf:35,1023|FS:0,0|Ov:50,50,100>
<Idle|MPos:0.000,0.000,0.000|Bf:35,1023|FS:0,0|Ov:40,50,100>
ok
ok
ok
ok
ok
<Run|MPos:0.000,0.000,-0.002|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,-0.006|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,-0.008|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,-0.011|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,-0.014|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,-0.017|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,-0.020|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,-0.023|Bf:34,1023|FS:4294967295,0|WCO:0.000,0.000,-1.384>
<Run|MPos:0.000,0.000,-0.027|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,-0.030|Bf:34,1023|FS:4294967295,0|Ov:40,50,100>
<Run|MPos:0.000,0.000,-0.033|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,-0.036|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,-0.039|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,-0.042|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,-0.045|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,-0.048|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,-0.052|Bf:34,1023|FS:4294967295,0>
<Run|MPos:0.000,0.000,-0.053|Bf:34,1023|FS:4294967295,0>

**Comment by terjeio** [comment]
> ... what caught my attention is FS:4294967295.0 from the beginning,

I need your settings to investigate this.

**Reply by BUGNOTTO** [reply]
Sure, how can I help? what settings do you need?

**Reply by terjeio** [reply]
[These](https://github.com/terjeio/ioSender/discussions/326#discussioncomment-6926940).

**Reply by BUGNOTTO** [reply]
Good morning friend, sorry for the delay, I was away from the company for health reasons. I'm returning to work today. so I will start testing again with the "Verbose" and "Show all responses in real time" option checked. Then I'll post the results log for you to analyze :)

I thank the attention

**Comment by BUGNOTTO** [comment]
New Error :/

This error happened when I clicked on the GRBL tab after defining the parameters in the Height Map tab

![erro 01](https://github.com/terjeio/ioSender/assets/99672754/e83f4545-f40c-4d08-97ef-800e28748647)

error log file attached

[erro 01.txt](https://github.com/terjeio/ioSender/files/12739964/erro.01.txt)

**Reply by BUGNOTTO** [reply]
hidden back error

![erro 02](https://github.com/terjeio/ioSender/assets/99672754/5458d50f-09f0-43e0-99c6-ef0fbc66ca4b)

**Comment by terjeio** [comment]
Is the last error repeatable? And there is a different error message behind the OutOf MemoryException box, what is/was it?

FYI there is a newer [edge version](http://www.io-engineering.com/downloads) available, can you try with that?

**Reply by BUGNOTTO** [reply]
interesting, I'll test it. Which file should I download and use?
ioSender XL Edge 2.0.44p4.zip or ioSender Edge 2.0.44p4.zip

**Reply by terjeio** [reply]
Looks like you are using the "normal" version: ioSender Edge 2.0.44p4.zip

**Comment by BUGNOTTO** [comment]
> Is the last error repeatable? And there is a different error message behind the OutOf MemoryException box, what is/was it?

I managed to simulate the error frequently, I think you will also be able to simulate it there. I believe this error is a bug, I have already updated to version 2.00.44p4 as you suggested.

First I import the .tap file, then I go to the "probing" tab, sub-tab "Tool length offset". I start the test to capture the zero point on the Z axis, after completing it I go to ava GRBL, when I do this this error occurs.

If I go directly to the "Height map" subtab and start the survey, this error does not occur.

I think the problem is leaving the survey tab and going to the GRBL tab after having already started the process of capturing the zero point of the z axis in the "tool length offset" tab

and the situation gets worse if I lower the feed rate values, reduce these values. :/

**Reply by BUGNOTTO** [reply]
Tomorrow I will record a video : )

**Reply by BUGNOTTO** [reply]
Error regarding polling speed in feed rate

https://github.com/terjeio/ioSender/assets/99672754/c3d5535b-79ef-4b6f-b634-37ac1452d746

**Reply by BUGNOTTO** [reply]
allocated 4GB in windows memory :/

https://github.com/terjeio/ioSender/assets/99672754/ce451230-e59f-4de4-a002-3dd42c43a12a

**Reply by BUGNOTTO** [reply]
failure during the polling process


https://github.com/terjeio/ioSender/assets/99672754/44fe8dc4-b1c1-4a43-8066-bd2bc4230d21

**Reply by terjeio** [reply]
This is due to the probe beeing triggered when moving to the next X position:

![image](https://github.com/terjeio/ioSender/assets/20260062/c6989771-4a8c-4ff3-9ab6-55d3ff7669fc)

You should add more retract motion? 2mm is not enough? The clearance between the last probe point and the retract position is only ~0.1mm.

_Alarm 4: Probe fail. Probe is not in the expected initial state before starting probe cycle when G38.2 and G38.3 is not triggered and G38.4 and G38.5 is triggered._

FYI you may add this option to the status report option to terminate the probing early if this happens:

![image](https://github.com/terjeio/ioSender/assets/20260062/86939228-a0b9-47c8-80f6-d98ecbbb4b05)

**Reply by BUGNOTTO** [reply]
Now I've ticked all the boxes :)

![image](https://github.com/terjeio/ioSender/assets/99672754/acf55d7d-f392-41f2-be74-7c629d475ea0)

stopped right at the beginning of the survey on a height map

https://github.com/terjeio/ioSender/assets/99672754/409b1ede-ce7c-4578-9fea-78d22b3a7667

**Reply by terjeio** [reply]
The probe was not triggered within the specified distance:

![image](https://github.com/terjeio/ioSender/assets/20260062/517ea934-2e80-44c1-b7e7-69d9472a5a7b)

The 1 micrometer difference is likely due to rounding of the reported values, internally full float precision is used. Increase the distance (DIstance de sonc).

**Comment by BUGNOTTO** [comment]
Tomorrow I'll try with the ioSender XL Edge 2.0.44p4.zip version

I even changed computers, I have another more powerful computer using Windows 10 Pro, where I recorded these videos above

**Reply by terjeio** [reply]
I do not expect the XL version to behave differently, only the layout differs - most of the code is shared (in dlls).

Post your controller settings, click _Ajuda > Sobre > To Clipboard_ and paste them into a comment. Your gcode file would be nice to have as well. I need to be able to replicate this bug if I ever is to track it down - and it could be in the controller?

**Reply by terjeio** [reply]
There is one oddity when you start probing the heigthmap - the other tabs are not disabled (greyed out) as they should. When you probe for the tool length offset they are. I am not able to replicate this and I cannot see where in the code it goes wrong, but obviously it does.

**Reply by BUGNOTTO** [reply]
Good Morning my friend.
I think this is what you are asking of me.

%
; grblHAL
; 1.1f.20230729
; [OPT:VNMSL2,35,1024,3,0]
; [NEWOPT:ENUMS,RT+,HOME,REBOOT,SED,RTC]
; [FIRMWARE:grblHAL]
; [NVS STORAGE:*FLASH]
; [DRIVER:ESP32]
; [DRIVER VERSION:230511]
; [DRIVER OPTIONS:v4.3.5]
; [BOARD:MKS Tinybee V1.0]
;
$N0=
$N1=
; 0 - Step pulse time
$0=10.0
; 1 - Step idle delay
$1=25
; 2 - Step pulse invert
$2=0
; 3 - Step direction invert
$3=7
; 4 - Invert stepper enable pin(s)
$4=7
; 5 - Invert limit pins
$5=7
; 6 - Invert probe pin
$6=1
; 8 - Ganged axes direction invert
$8=0
; 9 - PWM Spindle
$9=1
; 10 - Status report options
$10=511
; 11 - Junction deviation
$11=0.010
; 12 - Arc tolerance
$12=0.002
; 13 - Report in inches
$13=0
; 14 - Invert control pins
$14=6
; 15 - Invert coolant pins
$15=0
; 16 - Invert spindle signals
$16=0
; 17 - Pullup disable control pins
$17=1
; 18 - Pullup disable limit pins
$18=0
; 19 - Pullup disable probe pin
$19=0
; 20 - Soft limits enable
$20=0
; 21 - Hard limits enable
$21=0
; 22 - Homing cycle
$22=1
; 23 - Homing direction invert
$23=3
; 24 - Homing locate feed rate
$24=25.0
; 25 - Homing search seek rate
$25=500.0
; 26 - Homing switch debounce delay
$26=250
; 27 - Homing switch pull-off distance
$27=1.000
; 28 - G73 Retract distance
$28=0.100
; 29 - Pulse delay
$29=5.0
; 30 - Maximum spindle speed
$30=1000.000
; 31 - Minimum spindle speed
$31=0.000
; 32 - Mode of operation
$32=0
; 33 - Spindle PWM frequency
$33=5000
; 34 - Spindle PWM off value
$34=0.0
; 35 - Spindle PWM min value
$35=0.0
; 36 - Spindle PWM max value
$36=100.0
; 37 - Steppers deenergize
$37=0
; 39 - Enable legacy RT commands
$39=1
; 40 - Limit jog commands
$40=0
; 43 - Homing passes
$43=1
; 44 - Axes homing, first pass
$44=4
; 45 - Axes homing, second pass
$45=1
; 46 - Axes homing, third pass
$46=2
; 62 - Sleep enable
$62=0
; 63 - Feed hold actions
$63=2
; 64 - Force init alarm
$64=0
; 65 - Probing feed override
$65=0
; 100 - X-axis travel resolution
$100=640.200
; 101 - Y-axis travel resolution
$101=640.200
; 102 - Z-axis travel resolution
$102=640.200
; 110 - X-axis maximum rate
$110=500.000
; 111 - Y-axis maximum rate
$111=500.000
; 112 - Z-axis maximum rate
$112=500.000
; 120 - X-axis acceleration
$120=10.000
; 121 - Y-axis acceleration
$121=10.000
; 122 - Z-axis acceleration
$122=10.000
; 130 - X-axis maximum travel
$130=200.000
; 131 - Y-axis maximum travel
$131=200.000
; 132 - Z-axis maximum travel
$132=200.000
; 341 - Tool change mode
$341=0
; 342 - Tool change probing distance
$342=30.0
; 343 - Tool change locate feed rate
$343=25.0
; 344 - Tool change search seek rate
$344=200.0
; 345 - Tool change probe pull-off rate
$345=200.0
; 346 - Restore position after M6
$346=1
; 384 - Disable G92 persistence
$384=0
; 398 - Planner buffer blocks
$398=35
; 481 - Autoreport interval
$481=0
; 700 - Ativar eixo rotativo em Y
$700=0
%

**Reply by BUGNOTTO** [reply]
gcode

[test iosender cnc 27-09-23.zip](https://github.com/terjeio/ioSender/files/12765144/test.iosender.cnc.27-09-23.zip)

**Comment by terjeio** [comment]
Out of memory exception: I wonder if this is due to the resident guru in the ESP32 outputting garbage on connect:

![image](https://github.com/terjeio/ioSender/assets/20260062/27791fd4-72fb-40d2-86cc-fcb81668f9b9)

I see that you have _No action_ selected for _On connect_ in the sender connection dialog so this garbage should not appear.

![image](https://github.com/terjeio/ioSender/assets/20260062/62121f08-2acd-42f6-9fbf-99ddda964eb1)

I'll investigate this, it could be that I have to flush the input buffer on startup.

**Reply by BUGNOTTO** [reply]
error after applying the poll and switching from the poll tab to GRBL.

https://github.com/terjeio/ioSender/assets/99672754/10f5a0ea-6628-4c70-abca-a4f1a949c488

**Comment by BUGNOTTO** [comment]
I tried the XL version. The error appeared after finishing the surface scan, this time I didn't click on anything.

![image](https://github.com/terjeio/ioSender/assets/99672754/21945d40-a2b3-4921-91f7-967cfb5f96e8)

**Comment by terjeio** [comment]
I have upload a [new edge version](http://www.io-engineering.com/downloads) with some minor changes. However, after running numerous tests with your file I have yet to see any exceptions so it impossible for me to pin down where it goes wrong for you.
You might try to update the controller to a later version of grblHAL, but I doubt that would make any difference. Still worth a try?

**Reply by BUGNOTTO** [reply]
Good morning my friend. All good?
So how do I update the driver to a later version of grblHAL?

I saw that you updated these files.
ioSender Edge Keytest.zip
GCode Sender XL 2.0.44p5.zip
ioSender Edge 2.0.44p5.zip
![image](https://github.com/terjeio/ioSender/assets/99672754/07d30de4-59f8-47eb-8a0f-57e367d940df)

To update grblhal should I use any of these files?

grblHAL ESP32 Webui Cluster.zip
grblHAL_Due_Native_USB.ino.zip
grblHAL_Due_6axis_door_probe_openpnp.zip

**Reply by terjeio** [reply]
> So how do I update the driver to a later version of grblHAL?

Use the [Web Builder](http://svn.io-engineering.com:8080) - unless you have a custom board.

**Reply by BUGNOTTO** [reply]
My CNC controller board is the makerbase mks tinybee v1.0 , But I found it a bit advanced for me to update the grblHAL driver.
I will test version 2.0.44p5.zip

**Reply by BUGNOTTO** [reply]
New message, z axis is stopped, not moving. version GCode Sender XL 2.0.44p5

![image](https://github.com/terjeio/ioSender/assets/99672754/7be48c0b-da63-440e-a7db-4e363a8daf1b)

**Reply by terjeio** [reply]
I packaged my test sender by accident, delete _App.config_ in the sender folder or uncheck this setting:
![image](https://github.com/terjeio/ioSender/assets/20260062/3c62f488-bb5a-4ec4-8c8d-ccc9fdf99db9)

**Reply by BUGNOTTO** [reply]
canceled the survey before the cutter even touched the surface :/

![image](https://github.com/terjeio/ioSender/assets/99672754/38590a23-66ca-4a0e-b9d4-d37c1b0e28c7)
![image](https://github.com/terjeio/ioSender/assets/99672754/d4544d01-9bc9-4165-99e3-8c636e003754)

**Reply by terjeio** [reply]
Your probing distance (2mm) was too short.

**Reply by BUGNOTTO** [reply]
I increased the probing distance and the latch distance. and decreases feed rate speeds.
I'll keep trying here and I'll get back to you later.

![image](https://github.com/terjeio/ioSender/assets/99672754/bf355ae6-a01b-43ed-abc3-40be3c54a179)

**Reply by BUGNOTTO** [reply]
I've already redid the tests 7 times starting from scratch and the error no longer reproduced. I think it was the probing distance, now that I increased it to 4 mm there were no more errors.

**Comment by terjeio** [comment]
Good luck with Mach3 then.

**Reply by BUGNOTTO** [reply]
I want to help fix these problems, as io sender is much more complete. If you can guide me, I'm available.

**Reply by BUGNOTTO** [reply]
and I believe that here in Brazil there are more people with the same problem
